src/state.rs

```diff
@@ -100,7 +101,8 @@ impl ApplicationState {
     }
 
     pub fn disconnected_user(&mut self, endpoint: Endpoint) {
-        let user = self.lan_users.remove(&endpoint).unwrap();
+        let user = "???".to_string();
+        let user = self.lan_users.remove(&endpoint).unwrap_or(user);
```

Comment:
The tcp server can still be connected to from anything (apps other then termchat).

A simple way to reproduce  `nc ip tcp_port` when u ctrlc it will trigger that message.

Reply:
Ok, I am seeing that there is no paralellism between the `connected_user` and the `disconnected_user` functions. The `disconnected_user` function should be called only in case that the endpoint has been registered as user.

```rust
if state.is_logged_user(endpoint) {
    disconnected_user(endpoint);
}
```
Or something similar

Reply:
That makes sense. I'll remove the ??? then. If someone manages to trigger the panic there, we can always come back to this.